homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,368 @@
+            "manufacturer": "WiZ Light Platform",
+            "model": f"{self._bulbtype.name} Mac: {self._mac}",
+        }
+
+    def update_state_available(self):
+        """Update the state if bulb is available."""
+        self._state = self._light.status
+        self._available = True
+
+    def update_state_unavailable(self):
+        """Update the state if bulb is unavailable."""
+        self._state = False
+        self._available = False
+
+    async def update_state(self):
+        """Update the state."""
+        try:
+            await self._light.updateState()
+            if self._light.state is None:
+                self.update_state_unavailable()
+            else:
+                self.update_state_available()
+        except (TimeoutError, WizLightTimeOutError) as ex:
+            _LOGGER.debug(ex)
+            self.update_state_unavailable()
+        _LOGGER.debug(
+            "[wizlight %s] updated state: %s and available: %s",
+            self._light.ip,
+            self._state,
+            self._available,
+        )
+
+    def update_brightness(self):
+        """Update the brightness."""
+        if self._light.state.get_brightness() is None:
+            return
+        try:
+            brightness = self._light.state.get_brightness()
+            if 0 <= int(brightness) <= 255:
+                self._brightness = int(brightness)
+            else:
+                _LOGGER.error(
+                    "Received invalid brightness : %s. Expected: 0-255", brightness
+                )
+                self._brightness = None
+        # pylint: disable=broad-except
+        except Exception as ex:
+            _LOGGER.error(ex)
+            self._state = None
+
+    def update_temperature(self):
+        """Update the temperature."""
+        colortemp = self._light.state.get_colortemp()
+        if colortemp is None or colortemp == 0:
+            self._temperature = None
+            return
+
+        _LOGGER.debug(
+            "[wizlight %s] kelvin from the bulb: %s", self._light.ip, colortemp
+        )
+        temperature = color_utils.color_temperature_kelvin_to_mired(colortemp)
+        self._temperature = temperature
+
+    def update_color(self):
+        """Update the hs color."""
+        colortemp = self._light.state.get_colortemp()
+        if colortemp is not None and colortemp != 0:
+            self._hscolor = None
+            return
+        if self._light.state.get_rgb() is None:
+            return
+        try:
+            rgb = self._light.state.get_rgb()
+            if rgb[0] is None:
+                # this is the case if the temperature was changed
+                # do nothing until the RGB color was changed
+                return
+            warmwhite = self._light.state.get_warm_white()
+            if warmwhite is None:
+                return
+
+            self._hscolor = convertHSfromRGBCW(rgb, warmwhite)
+
+        # pylint: disable=broad-except
+        except Exception:
```

Comment:
What does this exception guard against?

Reply:
I'm not sure anymore about this. Maybe a relic from older times. I have to check where a broad exception can occur.